### PR TITLE
TT-6343 Fixing mem address pointer in prom custom metrics

### PR DESCRIPTION
**Pull request by tbuchaillot**
## Description

Basically, the for loop always took the last memory address of the custom metrics config. That makes the custom metrics always take the last custom metric value. 

Needs to add unit tests before merging.

## Related Issue

https://tyktech.atlassian.net/browse/TT-6343
## How This Has Been Tested

Added unit tests - tested manually

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). If pulling from your own
      fork, don't request your `master`!
- [ ] Make sure you are making a pull request against the **`master` branch** (left side). Also, you should start
      *your branch* off *our latest `master`*.
- [ ] My change requires a change to the documentation.
  - [ ] If you've changed APIs, describe what needs to be updated in the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] Modules and vendor dependencies have been updated; run `go mod tidy && go mod vendor`
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] Check your code additions will not fail linting checks:
  - [ ] `go fmt -s`
  - [ ] `go vet`

### CODE REVIEW

File: pumps/prometheus.go [hunk 1]
```diff
@@ -169,14 +169,32 @@ func (p *PrometheusPump) Init(conf interface{}) error {
...
 	if len(p.conf.CustomMetrics) > 0 {
 		customMetrics := []*PrometheusMetric{}
 		for i := range p.conf.CustomMetrics {
 			newMetric := &p.conf.CustomMetrics[i]
 			newMetric.aggregatedObservations = p.conf.AggregateObservations
 			errInit := newMetric.InitVec()
 			if errInit != nil {
-				p.log.Error(errInit)
+				return errInit
 			} else {
```
Comment (mativm02) [inline pumps/prometheus.go:198]: A very little observation: this else statement is useless, as you're returning the error before if it exists

**Review by mativm02 — APPROVED** [review]
LGTM

File: pumps/prometheus.go (hunk 1 above)
Comment (tbuchaillot) [reply pumps/prometheus.go]: solved!

File: pumps/prometheus.go [hunk 2]
```diff
@@ -197,6 +185,27 @@ func (p *PrometheusPump) Init(conf interface{}) error {
...
+//InitCustomMetrics initialise custom prometheus metrics based on p.conf.CustomMetrics and add them into p.allMetrics
+func (p *PrometheusPump) InitCustomMetrics() error {
+	if len(p.conf.CustomMetrics) > 0 {
+		customMetrics := []*PrometheusMetric{}
+		for i := range p.conf.CustomMetrics {
+			newMetric := &p.conf.CustomMetrics[i]
+			newMetric.aggregatedObservations = p.conf.AggregateObservations
+			errInit := newMetric.InitVec()
+			if errInit != nil {
+				return errInit
```
Comment (sredxny) [inline pumps/prometheus.go:197]: @tbuchaillot if we have an error is ok that we just return? and do not continue in the next iteration?

Comment (tbuchaillot) [reply pumps/prometheus.go:197]: @sredxny do you suggest that we just log the name of the invalid one and add the others?

Comment (tbuchaillot) [reply pumps/prometheus.go]: @sredxny changed to work in a non-blocking when missconfigured. good suggestion

**Review by sredxny — APPROVED** [review]
lgtm

**Pull request merged**